### PR TITLE
vmm: Fix Seccomp filter for AArch64

### PR DESCRIPTION
**Pull request by michael2012z**
Signed-off-by: Michael Zhao <michael.zhao@arm.com>

### CODE REVIEW

File: vmm/src/seccomp_filters.rs [hunk 1]
```diff
@@ -236,6 +246,10 @@ pub fn vmm_thread_filter() -> Result<SeccompFilter, Error> {
             allow_syscall(libc::SYS_fsync),
             #[cfg(target_arch = "x86_64")]
             allow_syscall(libc::SYS_ftruncate),
+            #[cfg(target_arch = "aarch64")]
+            // The definition of libc::SYS_ftruncate is missing on AArch64.
```
Comment (rbradford) [inline vmm/src/seccomp_filters.rs:250]: Please open an issue on the libc crate about this if there isn't one already.

Comment (michael2012z) [reply vmm/src/seccomp_filters.rs:250]: I created one for it: https://github.com/rust-lang/libc/issues/1803

File: vmm/src/seccomp_filters.rs [hunk 2]
```diff
@@ -120,6 +125,8 @@ fn create_vmm_ioctl_seccomp_rule() -> Result<Vec<SeccompRule>, Error> {
     Ok(or![
         and![Cond::new(1, ArgLen::DWORD, Eq, FIOCLEX)?],
         and![Cond::new(1, ArgLen::DWORD, Eq, FIONBIO)?],
+        and![Cond::new(1, ArgLen::DWORD, Eq, KVM_ARM_PREFERRED_TARGET,)?],
+        and![Cond::new(1, ArgLen::DWORD, Eq, KVM_ARM_VCPU_INIT,)?],
```
Comment (sboeuf) [inline vmm/src/seccomp_filters.rs:129]: These should have been gated by a condition based on the architecture.

Comment (michael2012z) [reply vmm/src/seccomp_filters.rs:129]: NP. I will create another commit to do it.

Comment (michael2012z) [reply vmm/src/seccomp_filters.rs:129]: Hmmm, checked other KVM APIs definitions, some other APIs are architecture specific as well: like KVM_GET_MSRS, KVM_GET_LAPIC. If these rules can distinguish with each other by it self (the const like `KVM_ARM_PREFERRED_TARGET` is  calculated by combining API ID, data structure size, etc.) and they don't affect each other, maybe we don't bother to gate them by arch-conditions. What do you think? @sboeuf

Comment (sboeuf) [reply vmm/src/seccomp_filters.rs:129]: I think we don't want potential guest to leverage the fact that some syscalls are available for other arch even when running on different architecture, as they might be able to trigger bugs and attack the host.
I agree that x86 specific syscalls should  also be gated properly.

Comment (michael2012z) [reply vmm/src/seccomp_filters.rs:129]: Agree, I will submit a PR to improve.
Thanks.

Comment (sboeuf) [reply vmm/src/seccomp_filters.rs:129]: Thanks @michael2012z

**Pull request merged**